packages/toolpad-app/src/toolpad/AppEditor/StructureExplorer/index.tsx

```diff
@@ -187,40 +213,74 @@ export default function PageStructureExplorer() {
 
   const handleNodeToggle = React.useCallback(
     (event: React.SyntheticEvent, nodeIds: string[]) => {
-      setExpandedDomNodeIds((prev) => {
-        const newIds = nodeIds.filter((id) => !prev.includes(id));
-        const retainedIds = prev.filter((id) => nodeIds.includes(id));
-        return [...retainedIds, ...newIds];
-      });
+      setExpandedDomNodeIds(nodeIds);
     },
     [setExpandedDomNodeIds],
   );
 
+  const handleNodeDelete = React.useCallback(
+    (event: React.KeyboardEvent<HTMLUListElement>) => {
+      // delete selected node if event.key is Backspace
+      if (event.key === 'Backspace') {
+        appStateApi.update(
```

Comment:
Seems like the correct logic, except we do may need to normalize the page row column sizes after deleting as done in the other places...
And now that this logic goes beyond just the `RenderOverlay` component, we should move all of this (deleteing node + oprhaned nodes + normalizing column sizes) to somewhere where we can reuse it more agnostically, not sure if that's easy or not.

Reply:
> And now that this logic goes beyond just the RenderOverlay component, we should move all of this (deleteing node + deleting orphaned nodes + normalizing column sizes) to somewhere where we can reuse it more 
I've added some changes to try and accomplish this, have a look